src/simularium/rendering/GBufferPass.ts

```diff
@@ -130,110 +130,144 @@ class GBufferPass {
         // y:-1 agent instance id (-1 so that 0 remains a distinct instance id from the background)
         // z:0 view space depth
         // alpha == -1 is a marker to discard pixels later, will be filled with frag depth
-        renderer.setClearColor(new Color(0.0, -1.0, 0.0), -1.0);
+        const COLOR_CLEAR = new Color(0.0, -1.0, 0.0);
+        const COLOR_ALPHA = -1.0;
+        renderer.setClearColor(COLOR_CLEAR, COLOR_ALPHA);
         renderer.setRenderTarget(colorBuffer);
+        renderer.autoClear = true;
 
-        // begin draw meshes
-        this.agentMeshGroup.visible = true;
-        this.agentFiberGroup.visible = true;
-        this.agentPDBGroup.visible = false;
-        this.instancedMeshGroup.visible = false;
-
-        scene.overrideMaterial = this.colorMaterialMesh;
-        renderer.render(scene, camera);
-        // end draw meshes
-
-        renderer.autoClear = false;
-
-        // draw instanced things
-        this.agentMeshGroup.visible = false;
-        this.agentFiberGroup.visible = false;
-        this.agentPDBGroup.visible = false;
-        this.instancedMeshGroup.visible = true;
-
-        scene.overrideMaterial = this.colorMaterialInstancedMesh;
-        renderer.render(scene, camera);
-        // end draw instanced things
-
-        // begin draw pdb
-        this.agentMeshGroup.visible = false;
-        this.agentFiberGroup.visible = false;
-        this.agentPDBGroup.visible = true;
-        this.instancedMeshGroup.visible = false;
-
-        scene.overrideMaterial = this.colorMaterialPDB;
-        renderer.render(scene, camera);
-        //end draw pdb
+        const DO_MESHES = true;
+        const DO_INSTANCED = true;
+        const DO_PDB = true;
+
+        if (DO_MESHES) {
+            // begin draw meshes
+            this.agentMeshGroup.visible = true;
+            this.agentFiberGroup.visible = true;
+            this.agentPDBGroup.visible = false;
+            this.instancedMeshGroup.visible = false;
+
+            scene.overrideMaterial = this.colorMaterialMesh;
+            renderer.render(scene, camera);
+            // end draw meshes
+            renderer.autoClear = false;
+        }
+
+        if (DO_INSTANCED) {
+            // draw instanced things
+            this.agentMeshGroup.visible = false;
+            this.agentFiberGroup.visible = false;
+            this.agentPDBGroup.visible = false;
+            this.instancedMeshGroup.visible = true;
+
+            scene.overrideMaterial = this.colorMaterialInstancedMesh;
+            renderer.render(scene, camera);
+            // end draw instanced things
+            renderer.autoClear = false;
+        }
+
+        if (DO_PDB) {
```

Comment:
can all three of these be true at the same time? `DO_MESHES` `DO_INSTANCED` and `DO_PDB`?

Reply:
Yes, the normal operation is to have them all true at the same time.  You would only switch them off if you were debugging, to try to isolate something. It was just really hard to do that, before this code change.